cpp/ql/lib/experimental/semmle/code/cpp/semantic/analysis/RangeAnalysisStage.qll

```diff
@@ -1190,19 +1190,27 @@ module RangeStage<DeltaSig D, BoundSig<D> Bounds, LangSig<D> LangParam, UtilSig<
     )
   }
 
+  /**
+   * Holds if `isLeft = true` and `mul`'s left operand is bounded by `delta`,
+   * or if `isLeft = false` and `mul`'s right operand is bounded by `delta`.
+   *
+   * If `upper = true` the computed bound is an upper boud, and if `upper = false`
+   * the computed bound is a lower bound. The `fromBackEdge`, `origdelta`, `reason`
```

Comment:
That's not quite true... Rather: If `upper = true` then the bound on the operand _contributes_ to an upper bound on `mul` - it is not itself necessarily an upper bound.

Reply:
Oh, yes. That's a good point. Will fix.

Reply:
Fixed in c14bccf7eb319beb636d2e0cf1d62d9b6fd51762 (I hope 🤞)